// @solarti/web3.js/src/utils/cluster.ts
const endpoint = {
  http: {
    devnet: 'http://api.devnet.solana.com',
    testnet: 'http://api.testnet.solana.com',
    'mainnet-beta': 'http://api.mainnet-beta.solana.com/',
    'devnet-slrt': 'http://api.arcaps.com',
    'testnet-slrt': 'http://api.testnet.arcaps.com',
    'mainnet-slrt': 'http://api.mainnet.arcaps.com',
  },
  https: {
    devnet: 'https://api.devnet.solana.com',
    testnet: 'https://api.testnet.solana.com',
    'mainnet-beta': 'https://api.mainnet-beta.solana.com/',
    'devnet-slrt': 'https://api.arcaps.com',
    'testnet-slrt': 'https://api.testnet.arcaps.com',
    'mainnet-slrt': 'https://api.mainnet.arcaps.com',
  },
};

export type Cluster = 'devnet' | 'testnet' | 'mainnet-beta' | 'devnet-slrt' | 'testnet-slrt' |'mainnet-slrt';

/**
 * Retrieves the RPC API URL for the specified cluster
 */
export function clusterApiUrl(cluster?: Cluster, tls?: boolean): string {
  const key = tls === false ? 'http' : 'https';

  if (!cluster) {
    return endpoint[key]['devnet-slrt'];
  }

  const url = endpoint[key][cluster];
  if (!url) {
    throw new Error(`Unknown ${key} cluster: ${cluster}`);
  }
  return url;
}
